### PR TITLE
vdirsyncer: 0.10.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by DamienCassou**
###### Motivation for this change

package update
###### Things done
- [x] Tested using sandboxing
- Built on platform(s)
  - [x] NixOS
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/vdirsyncer/default.nix [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 pythonPackages.buildPythonApplication rec {
-  version = "0.10.0";
+  version = "0.11.0";
   name = "vdirsyncer-${version}";
   namePrefix = "";
 
   src = fetchurl {
-    url = "https://pypi.python.org/packages/0b/fb/c42223e1e9169e4770194e62143d431755724b080d8cb77f14705b634815/vdirsyncer-0.10.0.tar.gz";
-    sha256 = "1gf86sbd6w0w4zayh9r3irlp5jwrzbjikjc0vs5zkdpa5c199f78";
+    url = "https://pypi.python.org/packages/39/e5/1e7097b5f0cd6de79ec9014f162a6000b77ca2a369ea8a1588a2eebff570/vdirsyncer-0.11.0.tar.gz";
```
Comment (matthiasbeyer) [inline pkgs/tools/misc/vdirsyncer/default.nix]: We could use the `version` variable here.

**Comment by matthiasbeyer** [comment]
LGTM. Care to add the thing I annotated?

File: pkgs/tools/misc/vdirsyncer/default.nix [hunk 2]
```diff
@@ -3,13 +3,13 @@
 # Packaging documentation at:
 # https://github.com/untitaker/vdirsyncer/blob/master/docs/packaging.rst
 pythonPackages.buildPythonApplication rec {
-  version = "0.10.0";
+  version = "0.11.0";
   name = "vdirsyncer-${version}";
   namePrefix = "";
```
Comment (FRidh) [inline pkgs/tools/misc/vdirsyncer/default.nix]: could you remove this as we're using `buildPythonApplication` which already removes the prefix.

File: pkgs/tools/misc/vdirsyncer/default.nix (hunk 1 above)
Comment (FRidh) [inline pkgs/tools/misc/vdirsyncer/default.nix]: can you use the pypi mirror.

Comment (FRidh) [inline pkgs/tools/misc/vdirsyncer/default.nix]: You can actually use old-style url's also.

**Comment by DamienCassou** [comment]
@matthiasbeyer @FRidh what about this new version?

khal, which depends on vdirsyncer, won't build (see #15625) but I guess this shouldn't prevent this PR from being merged.

**Pull request merged**